src/test/java/io/jenkins/plugins/artifact_manager_jclouds/s3/MinioIntegrationTest.java

```diff
@@ -0,0 +1,181 @@
+        minioServer = new GenericContainer("minio/minio")
+                .withEnv("MINIO_ACCESS_KEY", ACCESS_KEY)
+                .withEnv("MINIO_SECRET_KEY", SECRET_KEY)
+                .withCommand("server /data")
+                .withExposedPorts(port)
+                .waitingFor(new HttpWaitStrategy()
+                        .forPath("/minio/health/ready")
+                        .forPort(port)
+                        .withStartupTimeout(Duration.ofSeconds(10)));
+        minioServer.start();
+
+        Integer mappedPort = minioServer.getFirstMappedPort();
+        Testcontainers.exposeHostPorts(mappedPort);
+        minioServiceEndpoint = String.format("%s:%s", minioServer.getContainerIpAddress(), mappedPort);
+        
+        image = ArtifactManagerTest.prepareImage();
+    }
+    
+    @AfterClass
+    public static void shutDownClass() {
+        if (minioServer.isRunning()) {
+            minioServer.stop();
+        }
+    }
+    
+    @Before
+    public void setUp() throws IOException {
+        provider = new S3BlobStore();
+        CredentialsAwsGlobalConfiguration credentialsConfig = CredentialsAwsGlobalConfiguration.get();
+        credentialsConfig.setRegion(REGION);
+        CredentialsProvider.lookupStores(jenkinsRule.jenkins)
+                .iterator()
+                .next()
+                .addCredentials(Domain.global(), new AWSCredentialsImpl(CredentialsScope.GLOBAL, "MinioIntegrationTest", ACCESS_KEY, SECRET_KEY, "MinioIntegrationTest"));
+        credentialsConfig.setCredentialsId("MinioIntegrationTest");
+        
+        config = S3BlobStoreConfig.get();
+        config.setContainer(CONTAINER_NAME);
+        config.setPrefix(CONTAINER_PREFIX);
+        config.setCustomEndpoint(minioServiceEndpoint);
+        config.setUseHttp(true);
+        config.setUsePathStyleUrl(true);
+        config.setDisableSessionToken(true);
+        client = config.getAmazonS3ClientBuilderWithCredentials().build();
+        artifactManagerFactory = new JCloudsArtifactManagerFactory(provider);
+    }
+    
+    private void createBucketWithAwsClient(String bucketName) {
+        config.setContainer(bucketName);
+        client.createBucket(bucketName);
+    }
+    
+    @Test
+    public void canCreateBucket() throws Exception {
+        String testBucketName = "jenkins-ci-data";
+        Bucket createdBucket = config.createS3Bucket(testBucketName);
+        assertEquals(testBucketName, createdBucket.getName());
+        client.headBucket(new HeadBucketRequest(testBucketName));
+    }
+    
+    @Test
+    public void artifactArchive() throws Exception {
+        createBucketWithAwsClient("artifact-archive");
+        ArtifactManagerTest.artifactArchive(jenkinsRule, artifactManagerFactory, true, image);
+    }
+
+    @Test
+    public void artifactArchiveAndDelete() throws Exception {
+        createBucketWithAwsClient("artifact-archive-and-delete");
+        ReflectionTestUtils.setField(config, "deleteArtifacts", true);
```

Comment:
You should not need to do this. `JCloudsArtifactManagerTest` manages to avoid it, by having `CustomPrefixBlobStoreProvider` override `isDeleteArtifacts` and `isDeleteStashes`. Why can you not do the same here?

Reply:
I can do the same here